### PR TITLE
Reserved haskell keywords

### PR DESCRIPTION
**Pull request by nezbite**

### CODE REVIEW

File: IHP/IDE/SchemaDesigner/Controller/Tables.hs [hunk 1]
```diff
@@ -37,7 +37,7 @@ instance Controller TablesController where
         render NewTableView { .. }
 
     action CreateTableAction = do
-        let tableName = param "tableName"
+        let tableName = compileIdentifier (param "tableName")
```
Comment (mpscholten) [inline IHP/IDE/SchemaDesigner/Controller/Tables.hs:40]: IMO this should not be happen in the Schema Designer itself, because you can also edit the sql file manually and it should then still work. We should instead put it into the `IHP.SchemaCompiler`

File: IHP/IDE/SchemaDesigner/Compiler.hs [hunk 2]
```diff
@@ -76,12 +76,17 @@ compilePostgresType PTime = "TIME"
 compilePostgresType (PCustomType theType) = theType
 
 compileIdentifier :: _ -> Text
-compileIdentifier identifier = if identifierNeedsQuoting then tshow identifier else identifier
+compileIdentifier identifier = if isHaskellKeyword
```
Comment (mpscholten) [inline IHP/IDE/SchemaDesigner/Compiler.hs:79]: The sql field name should still be the "normal" thing. Only the haskell field should be named differently.

Comment (nezbite) [reply IHP/IDE/SchemaDesigner/Compiler.hs]: Haskell Keywords are fixed in the parser now. Manual SQL changes will now work

File: IHP/IDE/SchemaDesigner/View/Layout.hs [hunk 3]
```diff
@@ -76,10 +76,10 @@ databaseControls = [hsx|
...
 findStatementByName statementName statements = find pred statements
     where
-        pred CreateTable { name } | (toUpper name) == (toUpper statementName) = True
-        pred CreateTable { name } | (toUpper name) == (toUpper (tshow statementName)) = True
-        pred CreateEnumType { name } | (toUpper name) == (toUpper statementName) = True
-        pred CreateEnumType { name } | (toUpper name) == (toUpper (tshow statementName)) = True
+        pred CreateTable { name } | (compileIdentifier name) == (compileIdentifier statementName) = True
+        pred CreateTable { name } | (parseKeyword name) == (parseKeyword statementName) = True
+        pred CreateEnumType { name } | (compileIdentifier name) == (compileIdentifier statementName) = True
+        pred CreateEnumType { name } | (parseKeyword name) == (parseKeyword statementName) = True
```
Comment (mpscholten) [inline IHP/IDE/SchemaDesigner/View/Layout.hs:82]: Why these changes?

File: IHP/NameSupport.hs [hunk 4]
```diff
@@ -80,8 +80,71 @@ lcfirst = applyFirst toLower
...
+
+-- | Add '_' to the end of a name if it is a reserved haskell keyword
+--
+-- >>> parseKeyword "test"
+-- "test"
+--
+-- >>> parseKeyword "type"
+-- "type_"
+parseKeyword :: Text -> Text
+parseKeyword name = if toLower name `elem` haskellKeywords then name <> "_" else name
```
Comment (mpscholten) [inline IHP/NameSupport.hs:97]:
```suggestion
escapeHaskellKeyword :: Text -> Text
escapeHaskellKeyword name = if toLower name `elem` haskellKeywords then name <> "_" else name
```

File: IHP/NameSupport.hs [hunk 5]
```diff
@@ -80,8 +80,71 @@ lcfirst = applyFirst toLower
...
 ucfirst :: Text -> Text
 ucfirst = applyFirst toUpper
 {-# INLINE ucfirst #-}
+
+-- | Add '_' to the end of a name if it is a reserved haskell keyword
+--
+-- >>> parseKeyword "test"
+-- "test"
+--
+-- >>> parseKeyword "type"
```
Comment (mpscholten) [inline IHP/NameSupport.hs:94]:
```suggestion
-- >>> escapeHaskellKeyword "type"
```

File: IHP/NameSupport.hs [hunk 6]
```diff
@@ -80,8 +80,71 @@ lcfirst = applyFirst toLower
...
--- >>> ucfirst "Alread uppercase"
+-- >>> ucfirst "Already uppercase"
 -- "Already uppercase"
 ucfirst :: Text -> Text
 ucfirst = applyFirst toUpper
 {-# INLINE ucfirst #-}
+
+-- | Add '_' to the end of a name if it is a reserved haskell keyword
+--
+-- >>> parseKeyword "test"
```
Comment (mpscholten) [inline IHP/NameSupport.hs:91]:
```suggestion
-- >>> escapeHaskellKeyword "test"
```

File: IHP/IDE/SchemaDesigner/View/Layout.hs (hunk 3 above)
Comment (nezbite) [reply IHP/IDE/SchemaDesigner/View/Layout.hs:82]: This was necessary when the escaping was done in the parser. I'll revert this.
